### PR TITLE
[dv] Replace fileset_partner flag with fileset_ast flag

### PR DESCRIPTION
**Pull request by sha-ron**
Signed-off-by: Sharon Topaz <sharon.topaz@nuvoton.com>
This PR replaces the fileset_partner with fileset_ast for ast core.
It is required to enable to work in top level with the foundry database (that requires fileset_partner for the padring) and the open source AST.

### CODE REVIEW

**Comment by arnonsha** [comment]
Thanks @sha-ron 
This will require a change in my release script as well. Will check and let you know.

**Review by sriyerg — APPROVED** [review]
LGTM - should be easy to set this flag in your internal dvsim hjson file.

**Review by msfschaffner — APPROVED** [review]
Thanks @sha-ron. I will merge the change on the foundry side as well.

**Comment by arnonsha** [comment]
Michael
I did not approve the change just yet
It breaks my telease process.
Why did you merge befor my approval ?

**Comment by msfschaffner** [comment]
My apologies @arnonsha, I thought you were going to add the required flag to the release script in the foundry database once this has landed on both the OS and foundry side.

Let me know if it does not work and we need to revert the change.

**Comment by arnonsha** [comment]
Please revert for now.
Need to discuss with @sha-ron. Ill update when we both agree on all the details.

**Comment by msfschaffner** [comment]
Ok reverting it here: https://github.com/lowRISC/opentitan/pull/9794

Sorry about that.

**Comment by msfschaffner** [comment]
Quick update about this: After talking to @sha-ron and @arnonsha we've decided to stick with just one flag for now, but remove the padring module in the foundry database, since that one is not really used today (we only use the OS and the Nuvoton internal versions).

That way, the padring becomes like AST - there is a version in the open-source that we use for synthesis, and there is a Nuvoton internal one.

**Pull request merged**
